### PR TITLE
Webpack frontend proxy server

### PR DESCRIPTION
**Pull request by lneves12**
## Things to be aware while testing

The notification paths were updated so you need to redeploy you backend server.
If connecting to a ref-server the hot reload will only work on the page subscription channels. The notification pages suffered some path changes so it will only work after this being merged and the backend jar is redeployed on the ref servers.

## What does this PR change?

New frontend proxy server that serve all the static files and proxy everything for a chosen backend server.

This is based on my hackweek project work: https://hackweek.suse.com/17/projects/suse-manager-self-sufficient-frontend-server

**Example to run the proxy server targeting a local instance:**

"yarn proxy --server "https://server.tf.local" 

by default https://suma-refhead-srv.mgr.suse.de/ will be used

**Advantages:**

- Enable react hot reload of changed javascript
- No need to sync static files through lsyncd to our vm, they are served directly from the webpack proxy server
- Being able to change and troubleshoot frontend code connected to different backend servers, eg manager.suse.de, testesuite server, ref-servers, etc

- [x] **DONE**

## Documentation
- No documentation needed: **add explanation. 

- [x] **DONE**

## Test coverage
- No tests: **add explanation**

- [x] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/6000

- [x] **DONE**

### CODE REVIEW

**Comment by lneves12** [comment]
gitarro no changelog needed !!!

File: web/html/src/manager/virtualization/guests/list/guests-list.renderer.js [hunk 1]
```diff
@@ -1,23 +1,25 @@
-// @flow
-
-const ReactDOM = require('react-dom');
-const React = require('react');
+/* global module */
```
Comment (cbosdo) [inline web/html/src/manager/virtualization/guests/list/guests-list.renderer.js]: Same here for the `// @flow` comment

File: web/html/src/manager/virtualization/guests/edit/guests-edit.renderer.js [hunk 2]
```diff
@@ -1,21 +1,23 @@
-// @flow
-
-const React = require('react');
-const ReactDOM = require('react-dom');
+/* global module */
```
Comment (cbosdo) [inline web/html/src/manager/virtualization/guests/edit/guests-edit.renderer.js]: Why has the `// @flow` comment disappeared?

File: web/html/src/build/webpack.config.vendors.js [hunk 3]
```diff
@@ -4,15 +4,19 @@ const vendors = require("../vendors/vendors");
 const LicenseCheckerWebpackPlugin = require("license-checker-webpack-plugin");
 const CleanWebpackPlugin = require('clean-webpack-plugin');
 
-function getWebpackConfig({generateLicenses}) {
+module.exports = (env, argv) => {
+
+  const  isProductionMode = argv.mode !== "development";
```
Comment (cbbayburt) [inline web/html/src/build/webpack.config.vendors.js]: All the other checks are like `argv && argv.foo ...` maybe this should be the same?

File: web/html/src/build/webpack.config.vendors.js [hunk 4]
```diff
@@ -25,22 +29,16 @@ function getWebpackConfig({generateLicenses}) {
...
     name: "vendors",
-    entry: {
-      vendors
-    },
-    output:  {
+    entry: {vendors},
+    output: {
       path: path.resolve(__dirname, "../dist"),
-      filename: "vendors/vendors.bundle.js",
+      filename : "vendors/vendors.bundle.js",
```
Comment (cbbayburt) [inline web/html/src/build/webpack.config.vendors.js]:
```suggestion
      filename: "vendors/vendors.bundle.js",
```

File: web/html/src/.babelrc [hunk 5]
```diff
@@ -4,5 +4,10 @@
...
   ],
-  "plugins": ["@babel/plugin-proposal-class-properties", "@babel/plugin-transform-flow-strip-types", "@babel/plugin-syntax-object-rest-spread"]
-}
+  "plugins": [
+    "@babel/plugin-proposal-class-properties",
+    "@babel/plugin-transform-flow-strip-types",
+    "@babel/plugin-syntax-object-rest-spread",
+    "react-hot-loader/babel"
+  ]
+}
```
Comment (cbbayburt) [inline web/html/src/.babelrc]: Nitpick: no newline at the end

File: web/html/src/build.js [hunk 6]
```diff
@@ -1,9 +1,10 @@
 const shell = require('shelljs');
 const { fillSpecFile } = require("./build/fill-spec-file");
 
-const { code } = shell.exec("webpack --config build/webpack.config.js --mode production");
-if (code !== 0) {
-  shell.exit(code);
+const { code: codeVendors } = shell.exec("webpack --config build/webpack.config.vendors.js --mode production");
+const { code: codeSource } = shell.exec("webpack --config build/webpack.config.js --mode production");
+if (codeVendors !== 0 && codeSource !== 0) {
```
[the remark this answers is not in the document]
Comment (lneves12) [reply web/html/src/build.js]: yes! You are right! thanks

**Comment by lneves12** [comment]
@cbbayburt thanks, yes I will add all the instructions for the proxy and hot reload on https://github.com/SUSE/spacewalk/wiki/Frontend-Development-Environment----Uyuni-master

**Comment by cbbayburt** [comment]
Looks good now. Thanks for making our lives easier :smile:

**Comment by lneves12** [comment]
@ncounter  yes, my idea is to add it gradually every time we touch and move page to their own folder with a renderer file. I will add a new section on our wiki explaining how to activate the hot reload.

My long term plan is to have only folders inside "src/manager" where each folder match a feature with multiple pages inside. With that structure we can even deprecate the file "manager/index.js" which contains all the routes and auto-discover them adding an "index.js" file on the root of each folder exporting all the routes for that feature! Some kind of "plugin" system

@ncounter any suggestion of a better way to migrate all this or do you have another idea of how to structure our pages?

**Pull request merged**